### PR TITLE
ms1342: get distinct children bib levels

### PR DESCRIPTION
**Pull request by MingtaoSun**
@scoen @terenceingram 
http://ourweb.nla.gov.au/apps/jira/browse/DSCS-1342
- A query to get distinct children bib levels given the parent's work id

### CODE REVIEW

**Comment by terenceingram** [comment]
Don't really have a comment. It looks okay to me, but I am no expert in AmberDB.

File: src/amberdb/query/WorksQuery.java [hunk 1]
```diff
@@ -34,4 +43,25 @@
...
         return copies;
     }
+    
+    public static Set<BibLevel> getDistinctChildrenBibLevels(AmberSession sess, Long id){
+        List<byte[]> bibLevelCodes = new ArrayList<>();
+        try (Handle h = sess.getAmberGraph().dbi().open()) {
+            bibLevelCodes = h.createQuery(
+                    "SELECT DISTINCT p.value from property p, edge e where e.v_out=p.id \n"
+                            + "and e.txn_end = 0 and e.label = 'isPartOf' \n"
+                            + "and p.name = 'bibLevel' and p.txn_end = 0 and e.v_in = :workId; \n")
```
Comment (MingtaoSun) [inline src/amberdb/query/WorksQuery.java]: @scoen , I followed the existing examples to add \n at the end of each line. Were they actually necessary?

Comment (scoen) [inline src/amberdb/query/WorksQuery.java]: @MingtaoSun , no not necessary. Sometimes during development i'd print out queries to ensure they were well formed and it helped to format them for reading in those instances. That's the only reason.

File: src/amberdb/query/WorksQuery.java [hunk 2]
```diff
@@ -34,4 +44,27 @@
...
+    /**
+     * Find the distinct bib levels of all the children of the specified list of work ids
+     */
+    public static Set<BibLevel> getDistinctChildrenBibLevels(AmberSession sess, List<Long> workIds){
+        List<byte[]> bibLevelCodes = new ArrayList<>();
+        try (Handle h = sess.getAmberGraph().dbi().open()) {
+            bibLevelCodes = h.createQuery(
+                    "SELECT DISTINCT p.value from property p, edge e where e.v_out=p.id "
+                            + "and e.txn_end = 0 and e.label = 'isPartOf' "
+                            + "and p.name = 'bibLevel' and p.txn_end = 0 and e.v_in in ("+Joiner.on(",").join(workIds)+"); ")
```
Comment (MingtaoSun) [inline src/amberdb/query/WorksQuery.java]: @scoen , yep concatenation works!

**Pull request merged**